### PR TITLE
[50] [UI] As a logged in user, I can see Survey Question Screen

### LINKED ISSUE #50: [UI] As a logged in user, I can see Survey Question Screen
## Why

- We need a screen to show questions for the user to complete them.
 
## Acceptance Criteria

- Show question
- Show answer (with different type)
- Show quit survey button
- Show go to next question button
- Show `Submit` button on the last question

Screen Shot 2021-06-21 at 6 07 16 PM
Screen Shot 2021-06-21 at 6 07 30 PM

### PR DESCRIPTION
**Pull request by chiiida**
https://github.com/llleyelll/ic-surveys-ios/issues/50

## What happened 👀

- Implement Survey Question Screen UI.
 
## Insight 📝

- Mock data will be removed with integration.
 
## Proof Of Work 📹

https://user-images.githubusercontent.com/45258998/124729433-a4264200-df3a-11eb-8328-d3fcf9483534.mov

### CODE REVIEW

**Comment by blyscuit** [comment]
I think this is a big ticket. In the future, it might be better to break this down to parts for each of review:
- Scroll collection view
- Emoji cells
- Picker cell
- TextField cell

**Comment by blyscuit** [comment]
Also I think it's better if we subclass each type of cells here, would be easier to expand in the future, such as getting value from the cell.

**Comment by chiiida** [comment]
> Also I think it's better if we subclass each type of cells here, would be easier to expand in the future, such as getting value from the cell.

@blyscuit Sorry, I don't get your point. Could you please help clarify?
Do you mean to create a protocol or class something like `AnswerView`?

**Comment by chiiida** [comment]
> I think this is a big ticket. In the future, it might be better to break this down to parts for each of review:
> 
> * Scroll collection view
> * Emoji cells
> * Picker cell
> * TextField cell

@blyscuit I'll a new ticket for only answer view (choice, rating, text area, and nps), and this PR can focus only on Survey Question Screen Ui implementation. Would it be better this way? please help suggest 🙏

**Comment by blyscuit** [comment]
> @blyscuit Sorry, I don't get your point. Could you please help clarify?
> Do you mean to create a protocol or class something like `AnswerView`?

I think we can subclass `QuestionCollectionCell` or make a protocol like `QuestionCollectionCellProtocol` for each of the cell types. Then we don't need re-adding children logic. You could try first, if it doesn't seem promising we can discard the idea.

**Comment by chiiida** [comment]
> 
> I think we can subclass `QuestionCollectionCell` or make a protocol like `QuestionCollectionCellProtocol` for each of the cell types. Then we don't need re-adding children logic. You could try first, if it doesn't seem promising we can discard the idea.

I see. I'll try to improve that and also break this PR into smaller PRs as well. Thank you 🙇‍♀️

File: SurveyApp/Modules/SurveyQuestion/Cell/ChoiceCollectionCell.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    }
+    
+    override func configure(with question: SurveyQuestion) {
+        if question.pickType == SurveyQuestion.PickType.any {
+            answerView = MultipleChoiceAnswerView(answers: question.sortedAnswers)
+        } else {
+            answerView = ChoiceAnswerView(answers: question.sortedAnswers)
+        }
+        
+        super.questionLabel.text = question.text
```
Comment (blyscuit) [inline SurveyApp/Modules/SurveyQuestion/Cell/ChoiceCollectionCell.swift]: to reduce number of lines incase configure grows, can also
```suggestion
        super. configure(with question: question) 
```

**Review by blyscuit — APPROVED** [review]

File: SurveyApp/Modules/SurveyQuestion/Cell/ChoiceCollectionCell.swift (hunk 1 above)
Comment (chiiida) [reply SurveyApp/Modules/SurveyQuestion/Cell/ChoiceCollectionCell.swift]: I updated the code. Thank you for pointing that out 🙇‍♀️

**Review by markgravity — APPROVED** [review]
LGTM 🤖

**Pull request merged**